### DISCUSSION TITLE
Parametrize Catalog Dynamically

### DISCUSSION THREAD

**Discussion by vinialbert** [kedro-org/kedro#1650] [Idea]
Hello!

I've recently updated my projetcts to Kedro 0.18.1 and I'd like to understant if there is a new feature to Data Catalog that allows to dynamic parametrization of the inputs, either via kedro run or in notebook call. It will be very usefull to read groups of images without flood the entire catalog.

Today, in 0.17 version I change the hooks for the register config loader;

```python
def register_config_loader(
        self, conf_paths: Iterable[str], env: str, extra_params: Dict[str, Any],
    ) -> ConfigLoader:
        try:
            click_ctx = click.get_current_context(silent=True)
            if click_ctx is not None:
                if click_ctx.params.get("params") is not None:
                    return TemplatedConfigLoader(conf_paths, globals_pattern="*globals.yml", globals_dict={"run_id": click_ctx.params.get("params").get("run_id")})
                else:
                    return ConfigLoader(conf_paths)
            else:
                return TemplatedConfigLoader(conf_paths, globals_pattern="*globals.yml", globals_dict=extra_params)
        except:
                return ConfigLoader(conf_paths)
```

Is there any new feature in the new kedro release?

**Comment by noklam** [comment]
@vinialbert Could you provide an example of what does your function do? Are you trying to provide a runtime paramter that change the definition of  `catalog.yml`? If so this is already possible via Jinja template + `TemplateConfigLoader`.
